pkg/api/authz/mcpid.go

```diff
@@ -0,0 +1,38 @@
+package authz
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/obot-platform/nah/pkg/router"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+func (a *Authorizer) checkMCPID(req *http.Request, resources *Resources, user user.Info) (bool, error) {
+	if resources.MCPID == "" {
+		return true, nil
+	}
+
+	// MCP ID is the ID of either an MCPServer or an MCPServerInstance.
```

Comment:
nit: seems like this comment should be with the field definition on the `Resources` struct